op-service/txmgr/txmgr.go

```diff
@@ -196,8 +325,15 @@ func NewSimpleTxManager(name string, l log.Logger, cfg Config, backend ETHBacken
 	if cfg.NumConfirmations == 0 {
 		panic("txmgr: NumConfirmations cannot be zero")
```

Comment:
maybe return `error` instead?

Reply:
I'm fine leaving this as is for now. We have lots of config things to cleanup

Reply:
if it were the only remark in the ctor, yes, but there's also the chain id fetching, which should return an error here in the ctor instead of checking for a set chain id in every send call.

Reply:
Soooo change the `NewSimpleTxManager` method to return an error??

Reply:
It's only called in three places (1 test, batcher setup and proposer setup), so this wouldn't be too bad and would match the pattern we have in many ctors where we do some networking setup. Then we would also need to pass a `ctx`, which the call sites have though. 

But I looked at it again and have an even cleaner suggestion: add the `ChainID` to the `txmgr.Config`. It has multiple advantages
* Call sites know the id already or can query the id.
* Can remove the `chainID` from the `SimpleTxManager` state.
* Can remove `ChainID` from `ETHBackend`.
* Signature of `NewSimpleTxManager` doesn't need to be changed.

It's cleaner because the txmgr shouldn't need to query it itself and it can just require it as a config parameter.

In any case, setting up the chain id _should not_ be part of `CraftTx`. It belongs in setup code. It's a workaround to call it there just to not do other changes.

And as a general remark, it's totally ok in my opinion to change a function signature if that leads to a cleaner solution.